Homework/homework6/test_task1.py

```diff
@@ -0,0 +1,93 @@
+from collections import defaultdict
+
+from homework6.task1 import DeadlineError, Homework, HomeworkResult, Student, Teacher
+
+import pytest
+
+student = Student("Roman", "Petrov")
+teacher = Teacher("Daniil", "Shadrin")
+
+
+def test_homework_result_raise_typeerror():
+    homework_str = "homework"
+    with pytest.raises(TypeError, match="You gave not a Homework object"):
+        HomeworkResult(student, homework_str, "solution")
+
+
+def test_do_homework_for_expired_homework():
+    expired_homework = Homework("Any homework", 0)
+    with pytest.raises(DeadlineError, match="You are late"):
+        student.do_homework(expired_homework, "Any solution")
+
+
+def test_check_incorrect_homework():
+    homework = Homework("Any homework", 1)
+    incorrect_result = student.do_homework(homework, "<5")
+
+    assert teacher.check_homework(incorrect_result) is False
+    assert homework not in Teacher.homework_done
+
+
+def test_check_correct_homework():
+    Teacher.homework_done.clear()
+
+    homework = Homework("HW", 1)
+    correct_result = student.do_homework(homework, "More than 5 symbols.")
+
+    assert teacher.check_homework(correct_result) is True
```

Comment:
You code dependent on external varible.    If it is mutable it is better to use it as a fixture.  And in this case, do you realy need teacher?

Reply:
Okay about fixtures. But about teacher, yes, I need it here otherwise I cant use `check_homework`